src/agent/agent_manager.py

```diff
@@ -0,0 +1,279 @@
+#-
+# #%L
+# Contrast AI SmartFix
+# %%
+# Copyright (C) 2025 Contrast Security, Inc.
+# %%
+# Contact: support@contrastsecurity.com
+# License: Commercial
+# NOTICE: This Software and the patented inventions embodied within may only be
+# used as part of Contrast Security’s commercial offerings. Even though it is
+# made available through public repositories, use of this Software is subject to
+# the applicable End User Licensing Agreement found at
+# https://www.contrastsecurity.com/enduser-terms-0317a or as otherwise agreed
+# between Contrast Security and the End User. The Software may not be reverse
+# engineered, modified, repackaged, sold, redistributed or otherwise used in a
+# way not consistent with the End User License Agreement.
+# #L%
+#
+
+import subprocess
+import os
+
+from pathlib import Path
+from typing import Tuple, List
+
+from src import git_handler
+
+from src.utils import singleton, log, debug_log, error_exit, run_command
+from src.build_output_analyzer import extract_build_errors
+from src.contrast_api import FailureCategory, notify_remediation_failed
+#from src.telemetry_handler import update_telemetry
+
+from src.agent.agent_prompts import AgentPrompts
+from src.agent.agent_runner import AgentRunner
+
+
+@singleton
+class AgentManager:
+    def __init__(self, telemetry_handler):
+        debug_log("Initializing AgentManager")
+        self.telemetry_handler = telemetry_handler
+        self.agent_runner = AgentRunner(telemetry_handler)
+
+    def _build(self, remediation_id: str, build_command: str, repo_root: str) -> Tuple[bool, str]:
+        log(f"\n--- Running Build Command: {build_command} ---")
+        try:
+            # Use shell=True if the command might contain shell operators like &&, ||, > etc.
+            # Be cautious with shell=True if the command comes from untrusted input.
+            # Here, it's from an environment variable, assumed to be controlled.
+            result = subprocess.run(
+                build_command,
+                cwd=repo_root,
+                shell=True,
+                check=False, # Don't raise exception on non-zero exit
+                capture_output=True,
+                text=True,
+                encoding='utf-8', # Explicitly set encoding
+                errors='replace' # Handle potential encoding errors in output
+            )
+            self.telemetry_handler.update_telemetry("configInfo.buildCommandRunTestsIncluded", True);
+            output = result.stdout + result.stderr
```

Comment:
It's ok to just return the concatted output because it's logging? You might want to add something like:
"STDOUT:\n" + result.stdout + "\n\nSTDERR:\n" + result.stderr